test/core/client_channel/lb_policy/pick_first_test.cc

```diff
@@ -177,6 +269,61 @@ TEST_F(PickFirstTest, GoesIdleWhenConnectionFailsThenCanReconnect) {
   }
 }
 
+TEST_F(PickFirstTest, WithShuffle) {
+  testing::ScopedExperimentalEnvVar env_var(
+      "GRPC_EXPERIMENTAL_PICKFIRST_LB_CONFIG");
+  // 6 addresses have 6! = 720 permutations or 0.1% chance that the shuffle
+  // returns initial sequence
+  constexpr std::array<absl::string_view, 6> kAddresses = {
+      "ipv4:127.0.0.1:443", "ipv4:127.0.0.1:444", "ipv4:127.0.0.1:445",
+      "ipv4:127.0.0.1:446", "ipv4:127.0.0.1:447", "ipv4:127.0.0.1:448"};
+  absl::Status status = ApplyUpdate(
+      BuildUpdate(kAddresses, MakePickFirstConfig(true)), lb_policy_.get());
+  EXPECT_TRUE(status.ok()) << status;
+  std::vector<absl::string_view> prev_attempt_connect_order;
+  // There is 0.1% chance this check fails by design. Not an assert to prevent
+  // flake
+  if (absl::MakeConstSpan(prev_attempt_connect_order) ==
+      absl::MakeConstSpan(kAddresses)) {
+    gpr_log(GPR_INFO, "Address order did not change");
+  }
+  constexpr size_t kMaxAttempts = 5;
+  bool shuffled = false;
+  for (size_t attempt = 0; attempt < kMaxAttempts && !shuffled; ++attempt) {
+    std::vector<absl::string_view> address_order;
+    GetOrderAddressesArePicked(kAddresses, &address_order);
+    if (address_order != prev_attempt_connect_order) {
+      shuffled = true;
+    }
+    std::swap(prev_attempt_connect_order, address_order);
+  }
+  ASSERT_TRUE(shuffled) << "Addresses are not reshuffled";
+}
+
+TEST_F(PickFirstTest, ShufflingDisabled) {
+  constexpr std::array<absl::string_view, 6> kAddresses = {
+      "ipv4:127.0.0.1:443", "ipv4:127.0.0.1:444", "ipv4:127.0.0.1:445",
+      "ipv4:127.0.0.1:446", "ipv4:127.0.0.1:447", "ipv4:127.0.0.1:448"};
+  absl::Status status = ApplyUpdate(
+      BuildUpdate(kAddresses, MakePickFirstConfig(true)), lb_policy_.get());
+  EXPECT_TRUE(status.ok()) << status;
+  std::vector<absl::string_view> prev_attempt_connect_order;
```

Comment:
I don't think we need `prev_attempt_connect_order` at all in this test.  Instead, we can just check that the connection order in every iteration of the loop is the same as `kAddresses`:

```
for (size_t attempt = 0; attempts < 5; ++attempts) {
  std::vector<absl::string_vewi> address_order;
  GetOrderAddressesArePicked(kAddresses, &address_order);
  EXPECT_THAT(address_order, ::testing::ElementsAreArray(kAddresses));
}
```

Reply:
Done